builtin/logical/pki/path_acme_revoke.go

```diff
@@ -136,14 +136,14 @@ func (b *backend) acmeRevocationHandler(acmeCtx *acmeContext, _ *logical.Request
 func (b *backend) acmeRevocationByPoP(acmeCtx *acmeContext, userCtx *jwsCtx, cert *x509.Certificate, config *crlConfig) (*logical.Response, error) {
 	// Since this account does not exist, ensure we've gotten a private key
 	// matching the certificate's public key.
-	signer, ok := userCtx.Key.Key.(crypto.Signer)
+	public, ok := userCtx.Key.Key.(crypto.PublicKey)
```

Comment:
This comment is now very confusing.
I think what needs to happen is to check: "VerifyJWS(signature string)"  - that does seem to happen from the following call structure:
- acmeRevocationByPoP is called by: acmeRevocationHandler
- acmeRevocationHandler is called by patternAcmeRevoke
- patternAcmeRevoke wraps acmeRevocationHandler in acmeParsedWrappers
- acmeParsedWrappers calls ParseRequestParams
- ParseRequestParams calls VerifyJWS
- If VerifyJWS returns an error, then the call should be returned there.

This is confusing enough that I'd totally make a mistake with it.